app/src/main-process/menu/build-spell-check-menu.ts

```diff
@@ -0,0 +1,112 @@
+import { app, BrowserWindow, MenuItem } from 'electron'
+
+export async function buildSpellCheckMenu(
+  window: BrowserWindow | undefined
+): Promise<ReadonlyArray<MenuItem> | undefined> {
+  if (window === undefined) {
+    // window.webContents.off('context-menu', mergeDeferredContextMenuItems)
+    return
+  }
+
+  /*
+    When a user right clicks on a misspelled word in an input, we get event from
+    electron. That event comes after the context menu event that we get from the
+    dom.
+  */
+  return new Promise(resolve => {
+    window.webContents.once('context-menu', (event, params) =>
+      resolve(getSpellCheckMenuItems(event, params, window.webContents))
+    )
+  })
+}
+
+function getSpellCheckMenuItems(
```

Comment:
`getSpellCheckMenuItems` is the moved version of `mergeDeferredContextMenuItems`. But it doesn't merge items any more, but the logic for building them should be the same (asides from making MenuItems as opposed to IMenuItems now)